js/chart.js

```diff
@@ -0,0 +1,220 @@
+// Filter out an array with the users in past 60 minutes
+function filter(data) {
+  //this gives you an array of tuples where the first element is the hour and the last element is the # of users
+  var result = data.rows.map(
+    function(elem) {
+      return [elem[0], elem[1], elem[2]]
+    })
+
+  var date = new Date();
+  var hour = date.getHours();
+  var minute = date.getMinutes();
+  var x = [];
+
+  // sort result by hour and minute in ascending order
+  result.sort(function(a, b) {
+    var n = a[0] - b[0];
+    if(n !== 0) {
+      return n;
+    }
+    return parseInt(a[1]) - parseInt(b[1]);
+  });
+
+  // push number of pageviews in the past 60 minutes to an array
+  var counter = minute;
+  for(var i = 0; i < result.length; i++) {
+    // reset counter after 60 min mark
+    if(counter == 59) {
+      counter = 0;
+    }
+    if((result[i][0] == (hour-1)) && (result[i][1] >= counter)) {
+      x.push(Number(result[i][2]));
+      counter++;
+    }
+  }
+  // filtered array in past hour
+  return x;
+}
+
+var request = $.ajax({
+  type: 'GET',
+  url: ENDPOINTS.pageviews.hour,
+  dataType: 'jsonp',
+  //This only triggers if the ajax call was successful
+  //TODO: Define failure case
+  success: function(data) {
+    drawGraph(data);
+  }
+});
+
+function drawGraph(reply){
+  console.log('Here is the data from the ajax call')
+  console.log(reply)
+  console.log('now we can filter it!')
+  var yData = filter(reply)
+  var xData = []
+  console.log(xData)
+  console.log('and use it to draw the graph.')
+
+
+
+  for (var i = 0; i < yData.length; i++) {
+    xData.push(i);
+  }
+
+  var width = $('#pageviews-chart').width(),
+      height = $('#pageviews-chart').height();
+
+  var data = [ { label: "Page Views",
+                 x: xData,
+                 y: yData }, ] ;
+
+  var xy_chart = d3_xy_chart()
+      .width(width)
+      .height(height)
+      .xlabel("Time")
+      .ylabel("Page Views");
+
+  var svg = d3.select("#pageviews-chart").append("svg")
```

Comment:
this is the third reference to this ID string, ideally it would be a constant or parameter as opposed to a fragile magic string.